pkg/volume/cinder/cinder.go

```diff
@@ -193,31 +235,12 @@ func (plugin *cinderPlugin) newProvisionerInternal(options volume.VolumeOptions,
 	}, nil
 }
 
-func (plugin *cinderPlugin) getCloudProvider() (BlockStorageProvider, error) {
-	cloud := plugin.host.GetCloudProvider()
-	if cloud == nil {
-		if _, err := os.Stat(DefaultCloudConfigPath); err == nil {
-			var config *os.File
-			config, err = os.Open(DefaultCloudConfigPath)
-			if err != nil {
-				return nil, fmt.Errorf("unable to load OpenStack configuration from default path : %v", err)
-			}
-			defer config.Close()
-			cloud, err = cloudprovider.GetCloudProvider(openstack.ProviderName, config)
-			if err != nil {
-				return nil, fmt.Errorf("unable to create OpenStack cloud provider from default path : %v", err)
-			}
-		} else {
-			return nil, fmt.Errorf("OpenStack cloud provider was not initialized properly : %v", err)
-		}
-	}
-
-	switch cloud := cloud.(type) {
-	case *openstack.OpenStack:
-		return cloud, nil
-	default:
-		return nil, errors.New("invalid cloud provider: expected OpenStack")
+func (plugin *cinderPlugin) getCloudProvider(name string, namespace string) (BlockStorageProvider, error) {
+	cloud, err := openstack.NewOpenStack(plugin.host.GetKubeClient(), name, namespace)
```

Comment:
get secret here and pass into `NewOpenStack`

Reply:
Please see response in the comment on configFromSecret

Reply:
reconstructing the cloud provider here is a very different flow than any other plugin, and seems problematic for a few reasons:
* this is called frequently, and assumed to be low-cost (for example, constructing a client once per process, not once per cloudprovider access)
* it means the volume plugin cloudprovider is different than the controllermanager cloud provider, and has a different view of cloud provider resources. is that likely to cause problems doing things like finding instances, etc?